### PR TITLE
feat(connector): [OpenNode] Currency Unit Conversion

### PR DESCRIPTION
**Pull request by SagarDevAchar**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
- Addressing Issue: #2240 
- Modified two files in `hyperswitch/crates/router/src/connector/`
  - `opennode.rs`
    - Implement `get_currency_unit` function
    - Modify `ConnectorIntegration` implementations for `Opennode`
  - `opennode/transformers.rs`
    - Implement `OpennodeRouterData<T>` structure and functionality
    - Modify implementation for `OpennodePaymentsRequest`

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

We need to create a payment using opennode and test the currency unit from connector response and connector dashboard.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/opennode.rs [hunk 1]
```diff
@@ -72,6 +72,10 @@ impl ConnectorCommon for Opennode {
         "opennode"
     }
 
+    fn get_currency_unit(&self) -> api::CurrencyUnit {
+        api::CurrencyUnit::Base
+    }
```
Comment (SamraatBansal) [inline crates/router/src/connector/opennode.rs:77]: The CurrenyUnit will be `Minor` in this case as per open node requirements

```suggestion
        api::CurrencyUnit::Minor
    }
```

File: crates/router/src/connector/opennode/transformers.rs [hunk 2]
```diff
@@ -21,10 +52,12 @@ pub struct OpennodePaymentsRequest {
...
 
-impl TryFrom<&types::PaymentsAuthorizeRouterData> for OpennodePaymentsRequest {
+impl TryFrom<&OpennodeRouterData<&types::PaymentsAuthorizeRouterData>> for OpennodePaymentsRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
-        get_crypto_specific_payment_data(item)
+    fn try_from(
+        item: &OpennodeRouterData<&types::PaymentsAuthorizeRouterData>,
+    ) -> Result<Self, Self::Error> {
+        get_crypto_specific_payment_data(item.router_data)
```
Comment (SamraatBansal) [inline crates/router/src/connector/opennode/transformers.rs:60]: Pass the complete `item` Object to this function

```suggestion
        get_crypto_specific_payment_data(item)
```

Also in the function definition of ` get_crypto_specific_payment_data` change the argument from `&types::PaymentsAuthorizeRouterData` to `&OpennodeRouterData<&types::PaymentsAuthorizeRouterData>`

And there use `let amount = item.amount;` directly

**Review by SamraatBansal — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/opennode.rs (hunk 1 above)
Comment (SagarDevAchar) [reply crates/router/src/connector/opennode.rs]: Modified!

File: crates/router/src/connector/opennode/transformers.rs (hunk 2 above)
Comment (SagarDevAchar) [reply crates/router/src/connector/opennode/transformers.rs]: @SamraatBansal I have made the necessary modifications. 
Please let me know if I need to change anything else.

**Review by SamraatBansal — APPROVED** [review]
LGTM

**Review by ArjunKarthik — APPROVED** [review]

**Comment by deepanshu-iiitu** [comment]
Hey @SagarDevAchar  ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.
Thanks!

**Pull request merged**